### PR TITLE
Feature/synapse data mssql just reactive

### PR DESCRIPTION
**Pull request by whoswendy**
This PR contains code to add a synapse-data-mssql module that provides reactive support connecting to Microsoft SQL database.

### CODE REVIEW

File: data/data-samples/sample-data-mssql-reactive-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import io.americanexpress.synapse.data.mssql.entity.BaseEntity;
+
+/**
+ * The type Book entity.
+ */
+public class BookEntity extends BaseEntity {
+
+    private String title;
+
+    private String author;
```
Comment (andCastStatic) [inline data/data-samples/sample-data-mssql-reactive-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java]: May we add java docs here?

File: data/data-samples/sample-data-mssql-reactive-book/src/main/resources/schema.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+USE tempdb;
```
Comment (andCastStatic) [inline data/data-samples/sample-data-mssql-reactive-book/src/main/resources/schema.sql:12]: add copy right here

File: data/data-samples/sample-data-mssql-reactive-book/src/main/resources/schema.sql [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+);
+
+
+INSERT INTO book_entity (title, author, created_date_time, last_modified_date_time, created_by, last_modified_by, version)
+VALUES ('Synapse', 'Gabriel', GETDATE(), GETDATE(), 'John-Appleseed@email.com', 'John-Appleseed@email.com', 0);
+
+INSERT INTO book_entity (title, author, created_date_time, last_modified_date_time, created_by, last_modified_by, version)
+VALUES ('Revenge of Synapse', 'John', GETDATE(), GETDATE(), 'John-Appleseed@email.com', 'John-Appleseed@email.com', 0);
+
+GO
```
Comment (andCastStatic) [inline data/data-samples/sample-data-mssql-reactive-book/src/main/resources/schema.sql]: add line eof

**Review by andCastStatic — CHANGES_REQUESTED** [review]

File: data/data-samples/sample-data-mssql-reactive-book/src/main/resources/schema.sql (hunk 2 above)
Comment (whoswendy) [reply data/data-samples/sample-data-mssql-reactive-book/src/main/resources/schema.sql:12]: done

File: data/data-samples/sample-data-mssql-reactive-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java (hunk 1 above)
Comment (whoswendy) [reply data/data-samples/sample-data-mssql-reactive-book/src/main/java/io/americanexpress/data/book/entity/BookEntity.java]: done

**Pull request merged**